src/net_processing.cpp

```diff
@@ -1852,7 +1852,7 @@ void PeerManagerImpl::BlockConnected(const std::shared_ptr<const CBlock>& pblock
         LOCK(m_recent_confirmed_transactions_mutex);
         for (const auto& ptx : pblock->vtx) {
             m_recent_confirmed_transactions.insert(ptx->GetHash());
-            if (ptx->GetHash() != ptx->GetWitnessHash()) {
+            if (ptx->HasWitness()) {
```

Comment:
Comparing `Txid` and `Wtxid` is no longer allowed. We could use `ToUint256` or add a cache for `HasWitness`?

I like using `HasWitness` because it is immediately obvious what we are checking.

Reply:
Can `CTransaction::HasWitness` be changed to check whether `hash != m_witness_hash`?

Reply:
Could move the "iterate through" part into `ComputeWitnessHash`, and have `HasWitness` just return `hash.ToUint256() == m_witness_hash.ToUint256()`. Note that that's an example of where `ToUint256()` not doing a copy would make sense.